src/overlay_params.cpp

```diff
@@ -674,14 +698,18 @@ static void set_param_defaults(struct overlay_params *params){
 
 void
 parse_overlay_config(struct overlay_params *params,
-                  const char *env)
+                  const char *env, bool use_existing_preset)
 {
-   *params = {};
+   std::vector<int> default_preset = {-1, 0, 1, 2, 3, 4};
+   *params = {
+     .preset = use_existing_preset ? params->preset : default_preset
```

Comment:
Won't this be overwritten if preset exists in the config?

Reply:
Yes that's the idea